src/Plugin/WebformElement/MineOrganisationsData.php

```diff
@@ -0,0 +1,328 @@
+    }
+
+    if (FALSE !== $this->propertyAccessor->getValue($compositeElements, '[organisation_niveau_2][#access]')) {
+      $element['#organisation_niveau_2__value'] = $this->organisationHelper->getOrganisationEnhedNiveauTo($funktionsId);
+    }
+
+    if (FALSE !== $this->propertyAccessor->getValue($compositeElements, '[magistrat][#access]')) {
+      $element['#magistrat__value'] = $this->organisationHelper->getPersonMagistrat($funktionsId);
+    }
+  }
+
+  /**
+   * Updates basic sub elements.
+   */
+  private function updateBasicSubElements(&$element, $dataType) {
+    $brugerId = $this->getRelevantOrganisationUserId($dataType, FALSE);
+
+    if (NULL === $brugerId) {
+      return;
+    }
+
+    $compositeElements = $this->propertyAccessor->getValue($element, '[#webform_composite_elements]');
+
+    if (FALSE !== $this->propertyAccessor->getValue($compositeElements, '[name][#access]')) {
+      $element['#name__value'] = $this->organisationHelper->getPersonName($brugerId);
+    }
+
+    if (FALSE !== $this->propertyAccessor->getValue($compositeElements, '[email][#access]')) {
+      $element['#email__value'] = $this->organisationHelper->getPersonEmail($brugerId);
+    }
+
+    if (FALSE !== $this->propertyAccessor->getValue($compositeElements, '[az][#access]')) {
+      $element['#az__value'] = $this->organisationHelper->getPersonAZIdent($brugerId);
+    }
+
+    if (FALSE !== $this->propertyAccessor->getValue($compositeElements, '[phone][#access]')) {
+      $element['#phone__value'] = $this->organisationHelper->getPersonPhone($brugerId);
+    }
+
+    if (FALSE !== $this->propertyAccessor->getValue($compositeElements, '[location][#access]')) {
+      $element['#location__value'] = $this->organisationHelper->getPersonLocation($brugerId);
+    }
+  }
+
+  /**
+   * Fetches current user organisation user id.
+   */
+  private function getCurrentUserOrganisationId() {
+    $user = $this->entityTypeManager->getStorage('user')->load($this->account->id());
+
+    return $user->hasField('field_organisation_user_id') ? $user->get('field_organisation_user_id')->value : NULL;
+  }
+
+  /**
+   * Gets relevant organisation bruger or funktions id.
+   */
+  private function getRelevantOrganisationUserId(string $dataType, bool $returnFunktionsId) {
+    $currentUserId = $this->getCurrentUserOrganisationId();
+
+    if (NULL === $currentUserId) {
+      return NULL;
+    }
+
+    switch ($dataType) {
+      case self::DATA_DISPLAY_OPTION_CURRENT_USER:
+        return $currentUserId;
+
+      case self::DATA_DISPLAY_OPTION_MANAGER:
+        $managerInfo = $this->organisationHelper->getManagerInfo($currentUserId);
+
+        // @todo Handle multiple managers - for now just pick first one.
+        if ($returnFunktionsId) {
+          return reset($managerInfo)['funktionsId'];
+        }
+        else {
+          return reset($managerInfo)['brugerId'];
+        }
```

Comment:
Why don't we have separate function the different return types? Or let the caller pick the right key for `$managerInfo`?

Consider rewriting to 

```php
$managerInfo = reset($managerInfo);
return $returnFunktionsId ? $managerInfo)['funktionsId'] : $managerInfo['brugerId'];
```

or maybe even 

```php
…
return $managerInfo[$returnFunktionsId ? 'funktionsId' : 'brugerId'];
```

Reply:
Used the latter option.